### ISSUE TITLE
make rclone to use HTTP(s) proxy - useful in China

### ISSUE THREAD

**Issue by wutongskype** [kapitainsky/RcloneBrowser#50]
Windows version suggests adding a proxy server function

**Comment by wutongskype** [comment]
I am a Chinese, I need to break through GFW when using Google services. I hope to add a proxy server function to RcloneBrowser

**Comment by kapitainsky** [comment]
Rclone Browser is only GUI interface for rclone. You have to make sure that your rclone can connect where you want. Rclone Browser only uses network connectivity to check if there are new versions of software available. It does it by checking "latest release" on github. Github us not blocked in China so there no benefit in adding proxy to Rclone Browser

**Comment by kapitainsky** [comment]
From my experience in China the best way to go is good VPN.

**Comment by kapitainsky** [comment]
or use locally accessible storage providers e.g. rclone supports for example Alibaba Cloud (Aliyun) Object Storage System

**Comment by kapitainsky** [comment]
this might be help if you use shadowsocks or similar:

https://rclone.org/faq/#can-i-use-rclone-with-an-http-proxy

**Comment by danoe** [comment]
So right now the only way to set the proxy variable would be to set it system wide.
Might be nice to be able to pass ENV variables when RcloneBrowser runs rclone specifically.

**Comment by kapitainsky** [comment]
And why setting it globally is not good enough?

**Comment by danoe** [comment]
Then everything on the system will use the proxy, in his case only google should go through the proxy, as only that is being blocked by the GFW.

Browsers support this through PAC: https://superuser.com/questions/916281/how-to-force-browsers-to-use-proxy-server-for-specific-domains-such-as-google-co

Git supports passing proxy parameters to it: https://stackoverflow.com/questions/16067534/only-use-a-proxy-for-certain-git-urls-domains

Of course an alternative would be to ask rclone devs to add proxy parameters to rclone itself.

**Comment by kapitainsky** [comment]
Ok. Clear. I think I will talk to rclone guys how to do this in most clean way. Getting rid of GFW would be the easiest:)

**Comment by wutongskype** [comment]
> Ok. Clear. I think I will talk to rclone guys how to do this in most clean way. Getting rid of GFW would be the easiest:)
> […](#)
> On 4 Dec 2019, at 19:30, danoe ***@***.***> wrote: ﻿ Then everything on the system will use the proxy, in his case only google should go through the proxy, as only that is being blocked by the GFW. Browsers support this through PAC: https://superuser.com/questions/916281/how-to-force-browsers-to-use-proxy-server-for-specific-domains-such-as-google-co Git supports passing proxy parameters to it: https://stackoverflow.com/questions/16067534/only-use-a-proxy-for-certain-git-urls-domains Of course an alternative would be to ask rclone devs to add proxy parameters to rclone itself. — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or unsubscribe.

For me, when I use rclone alone, I will use this environment variable, but when I use rb, I don't know how to use the environment variable for proxy server.

> this might be help if you use shadowsocks or similar:
> 
> https://rclone.org/faq/#can-i-use-rclone-with-an-http-proxy


For me, when I use rclone alone, I will use this environment variable, but when I use rb, I don't know how to use the environment variable for proxy server.

**Comment by kapitainsky** [comment]
> For me, when I use rclone alone, I will use this environment variable, but when I use rb, I don't know how to use the environment variable for proxy server.

Could you please clarify this? Do you mean that when you set env variable for http proxy it works when you run rclone? but does not when you run rb?

**Comment by kapitainsky** [comment]
I have just run quick tests and when rclone is called from within rb it should see all environment variable you have set beforehand.

**Comment by kapitainsky** [comment]
@wutongskype  what you can do if you want http(s) proxies variables only set for rb is to start it via small bat file.

in the same directory as RcloneBrowser.exe file create run.bat file:

```
@ECHO OFF
set http_proxy=http://proxyserver:12345
set https_proxy=http://proxyserver:12345
START /B RcloneBrowser.exe
```
and start rb by running run.bat instead of exe file

**Comment by kapitainsky** [comment]
let me know if it works. If yes I will add http_proxy option to preferences in the next release. if not we will have to investigate it further.

**Comment by lvhe19880210** [comment]
> let me know if it works. If yes I will add http_proxy option to preferences in the next release. Then I can set this env variable for rb session only,
> 
> if not we will have to investigate it furtheir.

I am chinese user，it works~~thanks so much，please add http_proxy option to preferences in the next release

**Comment by kapitainsky** [comment]
good to have confirmation. I will include http_proxy options in the next release. In the meantime bat script will have to do its job:)

**Comment by wutongskype** [comment]
> @wutongskype 如果只希望为rb设置http（s）代理变量，可以通过小bat文件启动它。
> 
> 在与RcloneBrowser.exe文件相同的目录中，创建run.bat文件：
> 
> ```
> @ECHO OFF
> set http_proxy=http://proxyserver:12345
> set https_proxy=http://proxyserver:12345
> START /B RcloneBrowser.exe
> ```
> 
> 并通过运行run.bat而不是exe文件启动rb



> @wutongskype 如果只希望为rb设置http（s）代理变量，可以通过小bat文件启动它。
> 
> 在与RcloneBrowser.exe文件相同的目录中，创建run.bat文件：
> 
> ```
> @ECHO OFF
> set http_proxy=http://proxyserver:12345
> set https_proxy=http://proxyserver:12345
> START /B RcloneBrowser.exe
> ```
> 
> 并通过运行run.bat而不是exe文件启动rb



> @wutongskype what you can do if you want http(s) proxies variables only set for rb is to start it via small bat file.
> 
> in the same directory as RcloneBrowser.exe file create run.bat file:
> 
> ```
> @ECHO OFF
> set http_proxy=http://proxyserver:12345
> set https_proxy=http://proxyserver:12345
> START /B RcloneBrowser.exe
> ```
> 
> and start rb by running run.bat instead of exe file

RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。

RcloneBrowser has no problem with networking itself, but the Rclone program started through RcloneBrowser does not go through the proxy channel. If I used Rclone alone, I would use the variables in the .bat file using the method you described above. But when using RcloneBrowser, I don't know how to let Rclone started in the background of RcloneBrowser go through the proxy channel.

**Comment by wutongskype** [comment]
> > let me know if it works. If yes I will add http_proxy option to preferences in the next release. Then I can set this env variable for rb session only,
> > if not we will have to investigate it furtheir.
> 
> I am a chinese user，it works~~please add http_proxy option to preferences in the next release，thanks so much！
大佬你跟作者讲讲，我用RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。

**Comment by kapitainsky** [comment]
> RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。
> RcloneBrowser has no problem with networking itself, but the Rclone program started through RcloneBrowser does not go through the proxy channel. If I used Rclone alone, I would use the variables in the .bat file using the method you described above. But when using RcloneBrowser, I don't know how to let Rclone started in the background of RcloneBrowser go through the proxy channel.

When rclone is invoked by rb it uses whatever system wide proxy you have configured in http_proxy variable. If you don’t want to use system wide settings use bat file described earlier. In that case rb and rclone invoked by rb will use proxy defined in bat file. In the feature I will add this option to settings.

**Comment by lvhe19880210** [comment]
> > > let me know if it works. If yes I will add http_proxy option to preferences in the next release. Then I can set this env variable for rb session only,
> > > if not we will have to investigate it furtheir.
> > 
> > 
> > I am a chinese user，it works~~please add http_proxy option to preferences in the next release，thanks so much！
> > 大佬你跟作者讲讲，我用RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。

那就所有涉及翻墙的操作比如google云盘之类的都在broser里操作不就行了，broser没有的功能需要rclone命令行运行的本来就涉及不到翻墙吧，如果你实在想要命令行翻墙，那就像rclone作者说明的那样https://rclone.org/faq/#can-i-use-rclone-with-an-http-proxy在命令行后加入set httpxxxx这些命令，我亲测是有用的

**Comment by lvhe19880210** [comment]
> > RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。
> > RcloneBrowser has no problem with networking itself, but the Rclone program started through RcloneBrowser does not go through the proxy channel. If I used Rclone alone, I would use the variables in the .bat file using the method you described above. But when using RcloneBrowser, I don't know how to let Rclone started in the background of RcloneBrowser go through the proxy channel.
> 
> When rclone is invoked by rb it uses whatever system wide proxy you have configured in http_proxy variable. If you don’t want to use system wide settings use bat file described earlier. In that case rb and rclone invoked by rb will use proxy defined in bat file. In the feature I will add this option to settings.

i explian to him in Chinese again

**Comment by wutongskype** [comment]
> > > > let me know if it works. If yes I will add http_proxy option to preferences in the next release. Then I can set this env variable for rb session only,
> > > > if not we will have to investigate it furtheir.
> > > 
> > > 
> > > I am a chinese user，it works~~please add http_proxy option to preferences in the next release，thanks so much！
> > > 大佬你跟作者讲讲，我用RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。
> 
> 那就所有涉及翻墙的操作比如google云盘之类的都在broser里操作不就行了，broser没有的功能需要rclone命令行运行的本来就涉及不到翻墙吧，如果你实在想要命令行翻墙，那就像rclone作者说明的那样https://rclone.org/faq/#can-i-use-rclone-with-an-http-proxy 在命令行后加入set httpxxxx这些命令，我亲测是有用的

我需要用RcloneBrowser来下载GD网盘里面的资料，因为RB这样的图形界面比较方便。我在单独用Rclone的时候可以建一个批处理使用set httpxxxx这些命令，但是若是用RB的时候它调用的Rclone就没法指定这样的变量。我现在都是再用SSTAP这种类似VPN的全局代理软件，或者Proxifier这种可以劫持RB调用的Rclone进程的代理软件来使用，但是它不是很完美，会跟我机器上的其他程序冲突。

**Comment by lvhe19880210** [comment]
> > > > > let me know if it works. If yes I will add http_proxy option to preferences in the next release. Then I can set this env variable for rb session only,
> > > > > if not we will have to investigate it furtheir.
> > > > 
> > > > 
> > > > I am a chinese user，it works~~please add http_proxy option to preferences in the next release，thanks so much！
> > > > 大佬你跟作者讲讲，我用RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。
> > 
> > 
> > 那就所有涉及翻墙的操作比如google云盘之类的都在broser里操作不就行了，broser没有的功能需要rclone命令行运行的本来就涉及不到翻墙吧，如果你实在想要命令行翻墙，那就像rclone作者说明的那样https://rclone.org/faq/#can-i-use-rclone-with-an-http-proxy 在命令行后加入set httpxxxx这些命令，我亲测是有用的
> 
> 我需要用RcloneBrowser来下载GD网盘里面的资料，因为RB这样的图形界面比较方便。我在单独用Rclone的时候可以建一个批处理使用set httpxxxx这些命令，但是若是用RB的时候它调用的Rclone就没法指定这样的变量。我现在都是再用SSTAP这种类似VPN的全局代理软件，或者Proxifier这种可以劫持RB调用的Rclone进程的代理软件来使用，但是它不是很完美，会跟我机器上的其他程序冲突。

我亲测用bat方法启动broswer调用rclone的话直接就指定变量了，因此在使用broswer中gd有关的操作我这都没问题，但是如果直接用窗口命令行操作rclone的话就必须在窗口命令中手工加入set -http这类命令了

**Comment by lvhe19880210** [comment]
我用的不是sstap而是clash接管系统代理也不管用 ，但是bat启动设置让clash接管就ok了，你bat文件的http设置让ss-tap接管一下试试

**Comment by lvhe19880210** [comment]
> > > > > let me know if it works. If yes I will add http_proxy option to preferences in the next release. Then I can set this env variable for rb session only,
> > > > > if not we will have to investigate it furtheir.
> > > > 
> > > > 
> > > > I am a chinese user，it works~~please add http_proxy option to preferences in the next release，thanks so much！
> > > > 大佬你跟作者讲讲，我用RcloneBrowser 自身联网并无问题，但是通过RcloneBrowser启动的 Rclone程序没有走代理通道。如果我自己单独使用Rclone我会用你上面讲述的方法在.bat文件中使用变量。但是使用RcloneBrowser的时候我不知道该怎么让RcloneBrowser后台启动的Rclone走代理通道。
> > 
> > 
> > 那就所有涉及翻墙的操作比如google云盘之类的都在broser里操作不就行了，broser没有的功能需要rclone命令行运行的本来就涉及不到翻墙吧，如果你实在想要命令行翻墙，那就像rclone作者说明的那样https://rclone.org/faq/#can-i-use-rclone-with-an-http-proxy 在命令行后加入set httpxxxx这些命令，我亲测是有用的
> 
> 我需要用RcloneBrowser来下载GD网盘里面的资料，因为RB这样的图形界面比较方便。我在单独用Rclone的时候可以建一个批处理使用set httpxxxx这些命令，但是若是用RB的时候它调用的Rclone就没法指定这样的变量。我现在都是再用SSTAP这种类似VPN的全局代理软件，或者Proxifier这种可以劫持RB调用的Rclone进程的代理软件来使用，但是它不是很完美，会跟我机器上的其他程序冲突。

我用的不是sstap而是clash接管系统代理也不管用 ，但是bat启动设置让clash接管就ok了，你bat文件的http设置让ss-tap接管一下试试

**Comment by wutongskype** [comment]
> 我用的不是sstap而是clash接管系统代理也不管用 ，但是bat启动设置让clash接管就ok了，你bat文件的http设置让ss-tap接管一下试试

My Rclone bat
![Snipaste_2019-12-08_18-14-51](https://user-images.githubusercontent.com/40192806/70387978-d8e0b900-19e6-11ea-83ad-60210bd67932.jpg)

RB
![Snipaste_2019-12-08_18-15-47](https://user-images.githubusercontent.com/40192806/70387985-e8600200-19e6-11ea-8d7a-e1d81950313f.jpg)


如上图我在Bat里面可以set http，但是RcloneBrowser如何设置这个set http让它所调用的rclone来走代理通道呢？若是使用第三方代理程序来接管代理是可行方案，但是我的这台电脑里面还装了其他下载软件如Qbittorrent等BT软件，用SSTAP或者Proxifier都会影响其他程序，我打算把RcloneBrowser来当一个下载器来使用，所以想让RcloneBrowser调用的rclone能自己单独走代理通道。

**Comment by lvhe19880210** [comment]
> > 我用的不是sstap而是clash接管系统代理也不管用 ，但是bat启动设置让clash接管就ok了，你bat文件的http设置让ss-tap接管一下试试
> 
> My Rclone bat
> ![Snipaste_2019-12-08_18-14-51](https://user-images.githubusercontent.com/40192806/70387978-d8e0b900-19e6-11ea-83ad-60210bd67932.jpg)
> 
> RB
> ![Snipaste_2019-12-08_18-15-47](https://user-images.githubusercontent.com/40192806/70387985-e8600200-19e6-11ea-8d7a-e1d81950313f.jpg)
> 
> 如上图我在Bat里面可以set http，但是RcloneBrowser如何设置这个set http让它所调用的rclone来走代理通道呢？若是使用第三方代理程序来接管代理是可行方案，但是我的这台电脑里面还装了其他下载软件如Qbittorrent等BT软件，用SSTAP或者Proxifier都会影响其他程序，我打算把RcloneBrowser来当一个下载器来使用，所以想让RcloneBrowser调用的rclone能自己单独走代理通道。
我bat文件就这四行：
@ECHO OFF
set http_proxy=http://127.0.0.1:6170
set https_proxy=http://127.0.0.1:6170
START /B RcloneBrowser.exe
6170是我clash的http监听端口，你这里面18992是啥？
你的机场是http代理吗 ，如果是的话直接添机场的http代理和端口就不走代理软件了，如果是ss的话你不可能不经过ss客户端单独走机场代理啊

**Comment by wutongskype** [comment]
> > @echo OFF
> > set http_proxy=http://127.0.0.1:6170
> > set https_proxy=http://127.0.0.1:6170
> > START /B RcloneBrowser.exe
> > 6170是我clash的http监听端口，你这里面18992是啥？
> > 你的机场是http代理吗 ，如果是的话bat文件直接填机场的http代理和端口就不走代理软件了，如果是ss的话你不可能不经过ss客户端单独走机场代理啊
----------------------------
你还是没看明白我说的。你这个方法我试过了，用你上面的bat批处理的话就只有RcloneBrowser.exe这个走代理通道了，而它调用的rclone程序在后台没有走代理通道，如下图Rclone-GUI.exe，18992是我本地http代理的端口，我要是抛开RcloneBrowser单独用Rclone并设置Set http/https的话没问题的。

![Snipaste_2019-12-08_18-52-18](https://user-images.githubusercontent.com/40192806/70388382-dfbdfa80-19eb-11ea-8feb-daa9c2df9435.jpg)

**Comment by wutongskype** [comment]
> > rb如果用bat方法调用的话就等于已经设置了
> > 我bat文件就这四行：
> > @echo OFF
> > set http_proxy=http://127.0.0.1:6170
> > set https_proxy=http://127.0.0.1:6170
> > START /B RcloneBrowser.exe
> > 6170是我clash的http监听端口，你这里面18992是啥？
> > 你的机场是http代理吗 ，如果是的话bat文件直接填机场的http代理和端口就不走代理软件了，如果是ss的话你不可能不经过ss客户端单独走机场代理啊

---------------------------

如下图，所以右边文件列表显示不出来。。。我希望作者能在下个版本让RcloneBrowser能把Set Http这个环境变量的命令传递给它调用的rclone就完美了。
![Snipaste_2019-12-08_18-55-50](https://user-images.githubusercontent.com/40192806/70388416-74c0f380-19ec-11ea-8885-50af2db3035d.jpg)

**Comment by lvhe19880210** [comment]
> > > @echo OFF
> > > set http_proxy=http://127.0.0.1:6170
> > > set https_proxy=http://127.0.0.1:6170
> > > START /B RcloneBrowser.exe
> > > 6170是我clash的http监听端口，你这里面18992是啥？
> > > 你的机场是http代理吗 ，如果是的话bat文件直接填机场的http代理和端口就不走代理软件了，如果是ss的话你不可能不经过ss客户端单独走机场代理啊
> 
> 你还是没看明白我说的。你这个方法我试过了，用你上面的bat批处理的话就只有RcloneBrowser.exe这个走代理通道了，而它调用的rclone程序在后台没有走代理通道，如下图Rclone-GUI.exe，18992是我本地http代理的端口，我要是抛开RcloneBrowser单独用Rclone并设置Set http/https的话没问题的。
> 
> ![Snipaste_2019-12-08_18-52-18](https://user-images.githubusercontent.com/40192806/70388382-dfbdfa80-19eb-11ea-8feb-daa9c2df9435.jpg)

晕 你调用的是gui程序啊啊 都经过编译了不认系统变量很正常吧 那肯定不行啊 你应该直接命令行安装rclone，然后调用rclone二进制程序，用choco命令安装rclone就行

**Comment by lvhe19880210** [comment]
> > > rb如果用bat方法调用的话就等于已经设置了
> > > 我bat文件就这四行：
> > > @echo OFF
> > > set http_proxy=http://127.0.0.1:6170
> > > set https_proxy=http://127.0.0.1:6170
> > > START /B RcloneBrowser.exe
> > > 6170是我clash的http监听端口，你这里面18992是啥？
> > > 你的机场是http代理吗 ，如果是的话bat文件直接填机场的http代理和端口就不走代理软件了，如果是ss的话你不可能不经过ss客户端单独走机场代理啊
> 
> 如下图，所以右边文件列表显示不出来。。。我希望作者能在下个版本让RcloneBrowser能把Set Http这个环境变量的命令传递给它调用的rclone就完美了。
> ![Snipaste_2019-12-08_18-55-50](https://user-images.githubusercontent.com/40192806/70388416-74c0f380-19ec-11ea-8885-50af2db3035d.jpg)

你要用gui的话我觉得作者下个版本也是修改不好的。。。。

**Comment by wutongskype** [comment]
> > > > rb如果用bat方法调用的话就等于已经设置了
> > > > 我bat文件就这四行：
> > > > @echo OFF
> > > > set http_proxy=http://127.0.0.1:6170
> > > > set https_proxy=http://127.0.0.1:6170
> > > > START /B RcloneBrowser.exe
> > > > 6170是我clash的http监听端口，你这里面18992是啥？
> > > > 你的机场是http代理吗 ，如果是的话bat文件直接填机场的http代理和端口就不走代理软件了，如果是ss的话你不可能不经过ss客户端单独走机场代理啊
> > 
> > 
> > 如下图，所以右边文件列表显示不出来。。。我希望作者能在下个版本让RcloneBrowser能把Set Http这个环境变量的命令传递给它调用的rclone就完美了。
> > ![Snipaste_2019-12-08_18-55-50](https://user-images.githubusercontent.com/40192806/70388416-74c0f380-19ec-11ea-8885-50af2db3035d.jpg)
> 
> 你要用gui的话我觉得作者下个版本也是修改不好的。。。。

他能把变量传递给RB调用的rclone就行了。。。。

**Comment by kapitainsky** [comment]
Next version will have following preferences options available:

<img width="824" alt="Screenshot 2019-12-10 at 12 50 07" src="https://user-images.githubusercontent.com/15635386/70530886-bf0cb680-1b4b-11ea-994d-76ae50976e4d.png">

**Comment by kapitainsky** [comment]
@wutongskype, @lvhe19880210 

You can try it now in the beta release.

Windows 64 bits version is available here:

https://1drv.ms/u/s!Aq335pidOrBV3vpZJBG7m1zE1wotGA

and its signature

https://1drv.ms/u/s!Aq335pidOrBV3vpYryCkME6MunkX1A

**Comment by wutongskype** [comment]
> @wutongskype，@ lvhe19880210
> 
> 您现在可以在Beta版中试用它。
> 
> Windows 64位版本可在此处获得：[https](https://1drv.ms/u/s!Aq335pidOrBV3rx5D5qVv9s_7Z851g) :
> [//1drv.ms/u/s!Aq335pidOrBV3rx5D5qVv9s_7Z851g](https://1drv.ms/u/s!Aq335pidOrBV3rx5D5qVv9s_7Z851g)


--------------------------------------------------------------------------------
As shown in the figure, it still does not work, please see the list of GD files on the left is still not displayed
![Snipaste_2019-12-23_09-05-11](https://user-images.githubusercontent.com/40192806/71329864-53bbdf00-2564-11ea-84bc-0c53c79a83d0.jpg)

![Snipaste_2019-12-23_09-10-32](https://user-images.githubusercontent.com/40192806/71329869-56b6cf80-2564-11ea-8897-63de73d276c3.jpg)

**Comment by kapitainsky** [comment]
Because you defined proxy but not ticked the box “use manual configuration”. Regardless what you define you tell rb to use system settings.

**Comment by kapitainsky** [comment]
And don’t forget to restart rb after making changes

**Comment by kapitainsky** [comment]
assuming your proxy is http://192.168.1.100:18992 your proxy setting should be as folows:

<img width="628" alt="Screenshot 2019-12-23 at 09 01 45 1" src="https://user-images.githubusercontent.com/15635386/71347835-1dca2b00-2563-11ea-9c02-d400a00ecde8.png">

Let me know if it works.

To rule out issues with your proxy check from command line if you can connect to Google Drive. If not already done set proxy variable:

`set http_proxy="http://192.168.1.100:18992"`

then try to list Google Drive content:

`rclone ls remote:`

if this does not work it means that your proxy is not configured correctly. if this does work then we can investigate further.

You can find more about how rclone uses proxy here:

https://github.com/rclone/rclone/blob/master/docs/content/faq.md#can-i-use-rclone-with-an-http-proxy

**Comment by wutongskype** [comment]
> assuming your proxy is http://192.168.1.100:18992 your proxy setting should be as folows:
> 
> <img alt="Screenshot 2019-12-23 at 09 01 45 1" width="628" src="https://user-images.githubusercontent.com/15635386/71347835-1dca2b00-2563-11ea-9c02-d400a00ecde8.png">
> 
> Let me know if it works.
> 
> To rule out issues with your proxy check from command line if you can connect to Google Drive. If not already done set proxy variable:
> 
> `set http_proxy="http://192.168.1.100:18992"`
> 
> then try to list Google Drive content:
> 
> `rclone ls remote:`
> 
> if this does not work it means that your proxy is not configured correctly. if this does work then we can investigate further.
> 
> You can find more about how rclone uses proxy here:
> 
> https://github.com/rclone/rclone/blob/master/docs/content/faq.md#can-i-use-rclone-with-an-http-proxy

---------------------------------------------
It's incredible that I did n't see the option above 😂😂😂, this time RB she works normally and lists the file list of GD can also transfer data. thank you very much!

![Snipaste_2019-12-24_04-14-21](https://user-images.githubusercontent.com/40192806/71379662-12304000-2607-11ea-9c7b-a5d6ca7f9668.jpg)
![Snipaste_2019-12-24_04-30-56](https://user-images.githubusercontent.com/40192806/71379663-12c8d680-2607-11ea-9032-59e77a1c925f.jpg)
![Snipaste_2019-12-24_04-32-01](https://user-images.githubusercontent.com/40192806/71379664-12c8d680-2607-11ea-8d85-a59332ae3b0c.jpg)

**Comment by kapitainsky** [comment]
Great! Good it works now. Proxy configuration will be now included in all future versions

**Comment by plmxs2017** [comment]
> > 我用的不是sstap或者clash接管系统代理也不管用，但是bat启动设置让clash接管就ok了，你bat文件的http设置让ss-tap接管一下试试
> 
> 我的Rclone蝙蝠
> ![Snipaste_2019-12-08_18-14-51](https://user-images.githubusercontent.com/40192806/70387978-d8e0b900-19e6-11ea-83ad-60210bd67932.jpg)
> 
> RB
> ![Snipaste_2019-12-08_18-15-47](https://user-images.githubusercontent.com/40192806/70387985-e8600200-19e6-11ea-8d7a-e1d81950313f.jpg)
> 
> 如上图我在蝙蝠里面可以设置http，但是RcloneBrowser如何设置这个设置http让它所调用的rclone来走代理通道呢？如果是使用第三方代理程序来接管代理是可行方案，但是我的这台电脑里面还装了其他下载软件如Qbittorrent等BT软件，用SSTAP或Proxifier都会影响其他程序，我打算把RcloneBrowser来当一个下载器来使用，所以想让RcloneBrowser称为的rclone能自己单独走代理通道。

大佬你这个bat能否发我一份谢谢

**Comment by lvhe19880210** [comment]
> > > 我用的不是sstap或者clash接管系统代理也不管用，但是bat启动设置让clash接管就ok了，你bat文件的http设置让ss-tap接管一下试试
> > 
> > 
> > 我的Rclone蝙蝠
> > ![Snipaste_2019-12-08_18-14-51](https://user-images.githubusercontent.com/40192806/70387978-d8e0b900-19e6-11ea-83ad-60210bd67932.jpg)
> > RB
> > ![Snipaste_2019-12-08_18-15-47](https://user-images.githubusercontent.com/40192806/70387985-e8600200-19e6-11ea-8d7a-e1d81950313f.jpg)
> > 如上图我在蝙蝠里面可以设置http，但是RcloneBrowser如何设置这个设置http让它所调用的rclone来走代理通道呢？如果是使用第三方代理程序来接管代理是可行方案，但是我的这台电脑里面还装了其他下载软件如Qbittorrent等BT软件，用SSTAP或Proxifier都会影响其他程序，我打算把RcloneBrowser来当一个下载器来使用，所以想让RcloneBrowser称为的rclone能自己单独走代理通道。
> 
> 大佬你这个bat能否发我一份谢谢

新版可以设置proxy了 不需要bat了 ,preference里http/https proxy里设置成http(s)://127.0.0.1:xxxx ，xxxx为你clash的代理端口号

**Comment by plmxs2017** [comment]
谢谢！！看你的批处理不错，想学习学习！

lvhe19880210 <notifications@github.com> 于2020年1月22日周三 下午11:24写道：

**Comment by kapitainsky** [comment]
I will release new version with proxy support in the next week or two - I am travelling at the moment and don't have good internet access. With proxy supported by RB bat file should become obsolete. Until then it is very good workaround.

**Comment by kapitainsky** [comment]
It has been fixed in the [latest release](https://github.com/kapitainsky/RcloneBrowser/releases/tag/1.8.0)

This release has proxy options built in.
